src/components/ScaleLattice.vue

```diff
@@ -0,0 +1,346 @@
+import { kCombinations } from "xen-dev-utils";
+const props = defineProps<{
+  scale: Scale;
+  heldScaleDegrees: Set<number>;
+}>();
+
+type Coord = { x: number; y: number };
+type Node = {
+  x: number;
+  y: number;
+  label: string;
+  index: number;
+  fill: string;
+};
+
+type Coords = Coord[];
+
+const coords: Coords =
+  // Coords for every prime up to 23, indexes correspond to monzo indexes
+  // based on Kraig Grady's coordinate system https://anaphoria.com/wilsontreasure.html
+  [
+    {
+      x: 0,
+      y: 0,
+    },
+    {
+      x: 40,
+      y: 0,
+    },
+    {
+      x: 0,
+      y: -40,
+    },
+    {
+      x: 13,
+      y: -11,
+    },
+    {
+      x: -14,
+      y: -18,
+    },
+    {
+      x: -8,
+      y: -4,
+    },
+    {
+      x: -5,
+      y: -32,
+    },
+    {
+      x: 7,
+      y: -25,
+    },
+    {
+      x: 20,
+      y: -6,
+    },
+  ];
+
+const container = ref<HTMLDivElement | null>(null);
+
+const adjustLatticeSize = () => {
+  if (container.value) {
+    const containerBox = container.value.getBoundingClientRect();
+    const containerHeight = window.innerHeight - containerBox.top;
+    const image: HTMLElement | null = container.value.querySelector("#lattice");
+
+    if (container.value && image) {
+      image.style.width = `100%`;
+      image.style.height = `${containerHeight}px`;
+    }
+  }
+};
+
+onMounted(() => {
+  adjustLatticeSize();
+  window.addEventListener("resize", adjustLatticeSize);
+});
+
+onUnmounted(() => {
+  window.removeEventListener("resize", adjustLatticeSize);
+});
+
+// === Computed state ===
+
+const canLatticePrimes = computed(() => {
+  try {
+    return props.scale.intervals.reduce(
+      (canLattice: boolean, interval: Interval) =>
+        canLattice &&
+        interval.monzo
+          .toIntegerMonzo()
+          .slice(9) // hardcoded for now to the 23-limit
+          .reduce(
+            (isZero: boolean, component: number) => isZero && component === 0,
+            true
+          ),
+      true
+    );
+  } catch {
+    return false;
+  }
+});
+
+function maybeGetPrimeEquave(equave: Interval): number | null {
+  try {
+    const intergerMonzo: number[] = equave.monzo.toIntegerMonzo();
+    return intergerMonzo.reduce(
+      (
+        primeIndex: number | null,
+        component: number,
+        i: number
+      ): number | null => {
+        if (component === 1 && !primeIndex) {
+          return i;
+        }
+        if (component !== 0) {
+          throw new Error("Equave is not a prime number");
+        }
+        if (component === 0) {
+          return primeIndex;
+        }
+        return null;
+      },
+      null
+    );
+  } catch {
+    return null;
+  }
+}
+
+const equavePrimeIndex = computed(() =>
+  maybeGetPrimeEquave(props.scale.equave)
+);
+
+const nodes: ComputedRef<Node[] | string> = computed(() => {
+  try {
+    if (equavePrimeIndex.value !== 0) {
+      return "The lattice currently only supports scales with a 2/1 equave";
```

Comment:
This doesn't seem to trigger. I still get a messy visual for the scale [2/1, 3/1].